### PR TITLE
ETS-38 - GIAS API pt1

### PR DESCRIPTION
**Pull request by simonbu11**
Create API to provider learning provider from GIAS. Has api endpoint with tests to SOAP API layer.

TODO in pt2:
* Change AutoMapper for code mapping
* Implement SOAP API call for establishments
* Allow required fields to be specified

### CODE REVIEW

File: src/Dfe.Spi.GiasAdapter.Domain/LearningProvider.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace Dfe.Spi.GiasAdapter.Domain
+{
+    // TODO: This should come from the common data model nuget when ready
```
Comment (mmiddleton3301) [inline src/Dfe.Spi.GiasAdapter.Domain/LearningProvider.cs:3]: This exists and can be used now.

File: src/Dfe.Spi.GiasAdapter.Functions/LearningProviders/GetLearningProvider.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+namespace Dfe.Spi.GiasAdapter.Functions.LearningProviders
+{
+    public class GetLearningProvider
+    {
+        private const string FunctionName = nameof(GetLearningProvider);
+
+        private readonly ILearningProviderManager _learningProviderManager;
+        private readonly ILogger _logger;
+
+        public GetLearningProvider(ILearningProviderManager learningProviderManager, ILogger logger)
```
Comment (mmiddleton3301) [inline src/Dfe.Spi.GiasAdapter.Functions/LearningProviders/GetLearningProvider.cs:20]: Got a question re: the common logging around injecting here. Will chat to you in a bit!

**Review by mmiddleton3301 — CHANGES_REQUESTED** [review]
Just a couple of things, happy to chat about this.

File: src/Dfe.Spi.GiasAdapter.Domain/LearningProvider.cs (hunk 1 above)
Comment (simonbu11) [reply src/Dfe.Spi.GiasAdapter.Domain/LearningProvider.cs:3]: This will be coming an the next PR

**Review by mmiddleton3301 — APPROVED** [review]

**Pull request merged**